### PR TITLE
at86rf2xx: different types have RSSI base values

### LINKED ISSUE #4874: Using single RSSI_BASE_VALUE for all Atmel radios
Presently RSSI_BASE_VALUE is defined as -91 dBm in at86rf2xx.h. However, the [datasheets](http://www.atmel.com/products/wireless/802154/default.aspx?tab=documents) show some variation:

-91: RF230, RF231, RF232
-94: RF233 and SAMR21
-97 to -100: RF212 and RF212B (actual value depends on modulation)

### PR DESCRIPTION
**Pull request by OlegHahm**
Fixes #4874.

### CODE REVIEW

File: drivers/include/at86rf2xx.h [hunk 1]
```diff
@@ -87,7 +87,13 @@ extern "C" {
 /**
  * @brief   Base (minimal) RSSI value in dBm
  */
-#define RSSI_BASE_VAL                   (-91)
+#if MODULE_AT86RF233
+#   define RSSI_BASE_VAL                   (-94)
+#elif MODULE_AT86RF212B
+#   define RSSI_BASE_VAL                   (-97)
```
Comment (gebart) [inline drivers/include/at86rf2xx.h]: it seems like this value depends on the PHY mode, according to the data sheet for 212B.
The mode which yields -97 is not settable directly via `ifconfig x set page y`. IMO it's better to default to -98, which is what the modes for page 2 give on both EU and US channel bands.
Also, add a note that the 212B value is valid for page 2 only. The value is -100 or -99 for page 0, depending on channel setting.

**Comment by kb2ma** [comment]
Overall, the use of these separate values looks good enough to me. I can't comment on the 212B personally, but just to add a data point, let me ask someone I think is using this chip.

**Comment by OlegHahm** [comment]
Addressed your comments.

**Comment by kb2ma** [comment]
Sorry, no further information. I defer on the 212B.

File: drivers/include/at86rf2xx.h [hunk 2]
```diff
@@ -87,7 +87,17 @@ extern "C" {
...
  */
-#define RSSI_BASE_VAL                   (-91)
+#if MODULE_AT86RF233
+#   define RSSI_BASE_VAL                   (-94)
+#elif MODULE_AT86RF212B
+/**
+ * @note for the default settings in RIOT fur the at86rf212b,
+ *       for other seetings this value may change.
+ */
+#   define RSSI_BASE_VAL                   (-98)
```
Comment (PeterKietzmann) [inline drivers/include/at86rf2xx.h]: typo "fur" and unintended tap I guess

Comment (OlegHahm) [inline drivers/include/at86rf2xx.h]: Fixed typo, but which tap are you referring to?

Comment (PeterKietzmann) [inline drivers/include/at86rf2xx.h]: Wasn't a tap but spaces between "'#" and "define".

**Comment by PeterKietzmann** [comment]
Whoops, what hast the "twofish" stuff to do with this PR?

**Comment by OlegHahm** [comment]
Good question - rebase fixed it.

**Comment by PeterKietzmann** [comment]
@OlegHahm previously I was talking about the spaces between "'#" and "define". Besides of that please squash

**Comment by OlegHahm** [comment]
The indentation is on purpose because of the if-else. Squashed.

**Comment by PeterKietzmann** [comment]
Well, this indentation style for macros is new to me but I I won't stall this PR because of that.

**Comment by PeterKietzmann** [comment]
Ah, can't merge until @gebart approves.

**Comment by OlegHahm** [comment]
> Well, this indentation style for macros is new to me 

@PeterKietzmann, we use it rather frequently in RIOT. The reason is that the standard requires that the first character of the line for a preprocessor directive is the hash character.

**Pull request merged**